lib/ansible/modules/copy.py

```diff
@@ -636,7 +636,7 @@ def main():
                 # at this point we should always have tmp file
                 module.atomic_move(b_mysrc, dest, unsafe_writes=module.params['unsafe_writes'], keep_dest_attrs=not remote_src)
 
-            except (IOError, OSError):
+            except OSError:
                 module.fail_json(msg=f"Failed to copy {src!r} to {dest!r}.")
```

Comment:
This could benefit from the update as well:

```suggestion
            except OSError as ex:
                module.fail_json(msg=f"Failed to copy {src!r} to {dest!r}.", exception=ex)
```